src/components/Story-9-AddBook.jsx

```diff
@@ -0,0 +1,150 @@
+                headers: {
+                    "Content-Type": "application/json",
+                    "Accept": "application/json"
+                },
+                body : JSON.stringify(postBody)
+            })
+                .then(res => {
+                    if (res.status == 201) {
+                        setSuccess(true)
+                    }
+                    return res.json()
+            })
+                .then(data => {
+                    if (data.errors){
+                        setErrors(data.errors)
+                    }
+            })
+        } catch(err) {
+            console.error(err)
+        }
+    }
+
+    function createPostBody(){
+        let body = {}
+
+        if (title) body.title = title
+        if (author) body.author = author
+        if (genre !== null) body.genre_id = genre
+        if (!isNaN(year)) body.year = year
+        if (blurb) body.blurb = blurb
+        if (imageURL) body.image = imageURL
+
+        setPostBody(body)
+    }
+
+    function handleInput(e){
+        switch (e.target.id) {
+            case "title":
+                setTitle(e.target.value)
+                break;
+            case "author":
+                setAuthor(e.target.value)
+                break;
+            case "genre":
+                setGenre(parseInt(e.target.value))
+                break;
+            case "year":
+                setYear(parseInt(e.target.value))
+                break;
+            case "page_count":
+                setPageCount(e.target.value)
+                break;
+            case "image":
+                setImageURL(e.target.value)
+            case "blurb":
+                setBlurb(e.target.value)
+            default:
+                break;
+        }
+    }
+
+    useEffect(fetchGenres, [])
+    useEffect(createPostBody, [title, author, genre, year, pageCount, imageURL, blurb])
+
+    return (
+        <>
+            <div className="flex justify-center">
+                {!success && 
+                    <form onSubmit={submitBook} className="grid grid-cols-1 w-10/12 gap-2 md:w-4/12">
+                        <label htmlFor="title">Title (Required)</label>
+                        <input className="border-1 p-2 rounded" onChange={handleInput} type="text" id="title" name="title" />
+                        {errors.title && title === "" && <p className="text-red-600">{errors.title[0]}</p>}
+
+                        <label htmlFor="author">Author (Required)</label>
+                        <input className="border-1 p-2 rounded" onChange={handleInput} type="text" id="author" name="author" />
+                        {errors.author && author === "" && <p className="text-red-600">{errors.author[0]}</p>}
+
+                        <label htmlFor="genre">Genre (Required)</label>
+                        <select className="border-1 p-2 rounded" onChange={handleInput} id="genre">
+                            <option value={0} defaultValue>Any</option>
```

Comment:
In this case I think remove the any option, we should force the user to select a genre